test/format/format.sh

```diff
@@ -0,0 +1,301 @@
+#! /bin/bash
+
+[ -z $BASH_VERSION ] && {
+	echo "$0 is a bash script: \$BASH_VERSION not set, exiting"
+	exit 1
+}
+
+forcequit=0
+stop=0
+onintr()
+{
+	echo "$0: interrupted, cleaning up..."
+	forcequit=1
+	stop=1
+}
+trap 'onintr' 2
+
+usage() {
+	echo "usage: $0 [-aFSv] [-c config] "
+	echo "	  [-h home] [-j jobs] [-n runs] [-t minutes] [format-configuration]"
+	echo
+	echo "	  -a	       abort/recovery testing (defaults to off)"
+	echo "	  -c config    format configuration file (defaults to CONFIG.stress)"
+	echo "	  -F	       quit on first failure (defaults to off)"
+	echo "	  -h home      run directory (defaults to .)"
+	echo "	  -j jobs      parallel jobs (defaults to 8)"
+	echo "	  -n runs      number of runs (defaults to no limit)"
+	echo "	  -S	       run smoke-test configurations (defaults to off)"
+	echo "	  -t minutes   minutes to run (defaults to no limit)"
+	echo "	  -v	       verbose output (defaults to off)"
+
+	exit 1
+}
+
+# Smoke-tests.
+smoke_base_1="data_source=table rows=100000 threads=6 timer=4"
+smoke_base_2="$smoke_base_1 leaf_page_max=9 internal_page_max=9"
+smoke_list=(
+	"$smoke_base_1 file_type=fix"
+	"$smoke_base_1 file_type=row"
+	"$smoke_base_1 file_type=var"
+	"$smoke_base_1 file_type=row huffman_key=1 huffman_value=1"
+	"$smoke_base_1 file_type=var huffman_key=1 huffman_value=1"
+	"$smoke_base_1 file_type=row abort=1"
+	"$smoke_base_1 file_type=row data_source=lsm"
+	"$smoke_base_1 file_type=row statistics_server=1 rebalance=1"
+	"$smoke_base_2 file_type=var value_min=256"
+	"$smoke_base_2 file_type=row key_min=256"
+	"$smoke_base_2 file_type=row key_min=256 value_min=256"
+)
+smoke_next=0
+
+abort_test=0
+build=""
+config=""
+first_failure=0
+format_args=""
+home="."
+jobs=8
+minutes=0
+runs=0
+smoke_test=0
+verbose=0
+
+while :; do
+	case "$1" in
+	-a)
+		abort_test=1
+		shift ;;
+	-c)
+		config="$2"
+		shift ; shift ;;
+	-F)
+		first_failure=1
+		shift ;;
+	-h)
+		home="$2"
+		shift ; shift ;;
+	-j)
+		jobs="$2"
+		[[ '^[1-9]+$' =~ "$jobs" ]] && {
```

Comment:
I think the regular expression is not correct. 10 will be considered an illegal input.

Reply:
Thanks, fixed in 4ef126e.